mmcls/datasets/multi_label.py

```diff
@@ -9,19 +11,38 @@
 class MultiLabelDataset(BaseDataset):
     """Multi-label Dataset."""
 
+    def get_gt_labels(self):
+        """Get all ground-truth labels (categories).
+
+        Returns:
+            torch.Tensor: ground truth labels for all images.
+        """
+
+        return self.data_infos['all_gt_labels']
+
     def get_cat_ids(self, idx):
         """Get category ids by index.
 
         Args:
             idx (int): Index of data.
 
         Returns:
-            np.ndarray: Image categories of specified index.
+            torch.Tensor: Image categories of specified index.
         """
-        gt_labels = self.data_infos[idx]['gt_label']
-        cat_ids = np.where(gt_labels == 1)[0]
+        gt_label_index = self.data_infos['samples'][idx]['gt_label_index']
+        gt_label = self.data_infos['all_gt_labels'][gt_label_index]
+        cat_ids = torch.where(gt_label == 1)[0]
         return cat_ids
 
+    def prepare_data(self, idx):
+        results = copy.deepcopy(self.data_infos['samples'][idx])
+        gt_label_index = results.pop('gt_label_index')
+        results['gt_label'] = self.data_infos['all_gt_labels'][gt_label_index]
+        return self.pipeline(results)
```

Comment:
Modify `prepare_data` will influence all classes inherited from `MultiLabelDataset`, which will cause serious BC-breaking.
Consider creating a new class or use an option to disable this behavior by default?

Reply:
I now revert all changes on multi_label.py, and do this optimization only on voc.py. That will not affect datasets. Other developers may take voc.py as an example for optimization.